pkgs/tools/compression/zstd/default.nix

```diff
@@ -14,56 +15,50 @@ stdenv.mkDerivation rec {
     owner = "facebook";
   };
 
+  outputs = [ "out" "lib" "dev" ];
+
+  buildInputs = stdenv.lib.optional stdenv.isDarwin fixDarwinDylibNames;
+
+  nativeBuildInputs = [ cmake ]
```

Comment:
Other option would be to use meson. It should work with this magic:
https://github.com/NixOS/nixpkgs/blob/f373ecec8f49b24c3070ab4f66a680bee06980a5/pkgs/stdenv/adapters.nix#L63

Reply:
I just tried that. `enableShared` is still required though because the `preInstall` hook fails if shared don't get built, so this does not seem like a big win.